### DISCUSSION TITLE
How to find the best size of the sliding window used to calculate the distance matrix.

### DISCUSSION THREAD

**Discussion by sarala777** [TDAmeritrade/stumpy#787] [Q&A]
Hi Sean,

I need some help in finding the optimistic way to determine the size of the sliding window used to calculate the distance matrix.
The motifs vary when this parameter is different. And also how can to handle motifs of variable length. I have data from sensor that is captured every halfsecond continuously.

**Comment by seanlaw** [comment]
@sarala777 Thank you for your question and welcome to the STUMPY community. The short answer is, "there is simple way to determine the window size" as this single parameter is data/domain specific. As long as you are within the general ballpark (i.e., equal to or larger than the length of your event) then you should be able to capture your motif. However, the original matrix profile authors propose something called the "pan matrix profile", which describes a potential way to identify the best window size. Please read [this paper](https://www.cs.ucr.edu/~eamonn/PAN_SKIMP%20%28Matrix%20Profile%20XX%29.pdf) carefully.

It is still somewhat of a feature-in-progress but we have added this capability to STUMPY in the `stumpy.stimp` function. Please see the [API docs](https://stumpy.readthedocs.io/en/latest/api.html#stumpy.stimp) and this work-in-progress [tutorial](https://github.com/TDAmeritrade/stumpy/blob/main/docs/Tutorial_Pan_Matrix_Profile.ipynb).

Unfortunately, this is the limit of the support that we can provide on this topic as it is still an active area of research.
